### PR TITLE
Add default command to add spacer tiles to dock

### PR DESCRIPTION
**Pull request by ffittschen**
The `persistent-apps` key allows to add spacer tiles to the dock (see screenshot), by appending a specific string to the array of the key. 

This time I also tried to add some recording for this default key. I just faced one challenge: there is no `-array-remove` command, so that I can programmatically remove the spacer tile. That's why I described the manual steps in the `logRollbackInfo()`. It would be great it this can be automated though.. Do you happen to have an idea on how to achieve this?

However, this also affects the generated page for this command, as I don't think it should contain the `defaults delete com.apple.dock persistent-apps` command. (I didn't try to run it, but I guess it will remove all apps from the dock?). Is there any way to _not_ show the `delete` command for a key?

### TODOs

- [ ] Record real screenshots (the ones in this PR are just created with the `touch` command to be able to run the server)

## Screenshot

Screen Shot 2020-10-17 at 11 47 53@2x

### CODE REVIEW

**Comment by yannbertrand** [comment]
Wow. Nice job!

So here is what happens when you run the delete command:
When delete run

I guess you're right, we should handle arrays and that command especially differently.

Is there any chance you tried to save your entire Dock setup by saving your `defaults read com.apple.dock persistent-apps` output?

**Comment by ffittschen** [comment]
Hmm.. that's unfortunate. For the command in #46, the `delete` is fine, since it only stores _additional_ information, but deleting the whole dock setup is not great 😄

Saving and restoring with `defaults read` sounds like a good idea for recording the screenshots! But for the rendered website, we would probably need some advanced behavior, as we would add the spacers with `array-add`, but write a textual description of the manual steps to delete the spacers, as this is not possible with the `defaults` command. 

I'll try the saving + restoring during the weekend

**Comment by yannbertrand** [comment]
What if we provide a way to hide the "delete" section? Or add a markdown comment on it?

**Comment by yannbertrand** [comment]
So I've tried some stuff, what do you of having a custom alert before the delete command?

alert before running the delete command

We could display it by adding a `deleteAlert` section in the defaults.yml file:
```yml
        deleteAlert:
          type: danger
          message:
            "This command will remove all saved Apps from your Dock!<br>
            Make sure to save your config using
            ```bash
            defaults read com.apple.dock persistent-apps > dock-persistent-apps.txt
            ```
            You can restore it using
            ```bash
            defaults write ... TODO
            ```"
```

Let me know what you think!

**Comment by ffittschen** [comment]
I really like the customizable message above the delete command! This promotes transparency over hiding possibly destructive actions

File: defaults.yml [hunk 1]
```diff
@@ -171,6 +171,27 @@ categories:
...
+              width: 740
+              height: 463
+            text: Appends a spacer tile to the persistent apps area of the dock.
+          - value: "'{\"tile-type\"=\"small-spacer-tile\";}'"
+            image:
+              filename: "small-spacer-tile.png"
+              width: 740
+              height: 463
+            text: Appends a spacer tile to the persistent apps area of the dock.
+        versions: [Big Sur, Catalina, Mojave]
```
Comment (yannbertrand) [inline defaults.yml:193]:
```suggestion
        deleteAlert:
          type: danger
          message:
            "This command will remove all saved Apps from your Dock!<br>
            Make sure to save your config using<br>
            `defaults export com.apple.dock defaults-dock-backup.plist`<br>
            You can restore it using<br>
            `defaults import com.apple.dock defaults-dock-backup.plist`"
        versions: [Big Sur, Catalina, Mojave]
```

You should be able to use this now 🙂 

(See https://github.com/yannbertrand/macos-defaults/commit/71eb300dd6c62ca7fa78f8e1200c27b0824c6849 for more details on implementation)

**Comment by yannbertrand** [comment]
Thanks for your help, I'm finishing the job 🙂

**Comment by ffittschen** [comment]
Looks great, thank you for finishing it! And I also learned something new again because of this project: `defaults export` and `defaults import` 🥳

**Pull request merged**